src/schema/v2/partner/Settings/updatePartnerProfileImageMutation.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import { ResolverContext } from "types/graphql"
+import { ImageType } from "schema/v2/image"
+
+interface UpdatePartnerProfileImageInputProps {
+  profileId: string
+  type: string
+  remoteImageS3Key: string
+  remoteImageS3Bucket: string
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdatePartnerProfileImageSuccess",
+  isTypeOf: (data) => !!data.id,
+  fields: () => ({
+    image: {
```

Comment:
I _think_ you'd want `partner` here just to refresh the UI after the image is done uploaded. Totally fine!

Reply:
(Also kinda nice / easy conceptually -- just being like: i've updated some attribute of a thing; now return the thing -- because i might want to refresh something _else_ related to the thing after the attribute of the thing was updated, which is common. Things in things)

Reply:
Cool 👍 trying to think through how it would be possible to refresh the page without that ID, but I don't think its possible

Reply:
Re: this discussion, could even accept a `partnerID` as the input argument for the mutation instead of profile id. That might be more convenient (one less layer of abstraction from partner to profile that the client needs to know).

And/or, the gravity API could return the `profile/owner` back instead of or along with the image if returning just the 'image' from the gravity API isn't helpful.

Reply:
Ah `partnerID`, good idea, that sounds clearer that this, I'll adjust the api

Reply:
WDYT: https://github.com/artsy/gravity/pull/18852